tapchannel/auf_leaf_signer_test.go

```diff
@@ -131,6 +142,171 @@ func TestAuxLeafSigner(t *testing.T) {
+			channeldb.SimpleTaprootFeatureBit |
+			channeldb.TapscriptRootBit,
+		IsInitiator: true,
+	}
+	randInputProof := randProof(t)
+	commitTx := &randInputProof.AnchorTx
+	outgoingHtlcs := make(map[input.HtlcIndex][]*cmsg.AssetOutput)
+	outgoingHtlcs[0] = []*cmsg.AssetOutput{
+		cmsg.NewAssetOutput(
+			randInputProof.Asset.ID(), randInputProof.Asset.Amount,
+			randInputProof,
+		),
+	}
+
+	com := cmsg.NewCommitment(
+		nil, nil, outgoingHtlcs, nil, lnwallet.CommitAuxLeaves{},
+	)
+	cancelChan := make(chan struct{})
+
+	// Constructing multiple jobs will allow us to assert that later jobs
+	// are cancelled successfully.
+	numSigJobs := 10
+	jobs := make([]lnwallet.AuxSigJob, 0, numSigJobs)
+	outputIdxCounter := int32(0)
+
+	for range numSigJobs {
+		newJob := RandAuxSigJob(
+			t, cancelChan, lfn.Some[tlv.Blob](com.Bytes()),
+			outputIdxCounter,
+		)
+		jobs = append(jobs, newJob)
+		outputIdxCounter++
+	}
+
+	// We'll try to cancel all signature jobs once the aux signer completes
+	// a third of the batch.
+	cancelThreshold := numSigJobs / 3
+
+	err := signer.SubmitSecondLevelSigBatch(chanState, commitTx, jobs)
+	require.NoError(t, err)
+
+	select {
+	case <-time.After(testTimeout):
+		t.Fatalf("timeout waiting for response")
+	case <-jobs[cancelThreshold].Resp:
+		// Send the cancel signal; jobs at the end of the batch should
+		// not be processed.
+		close(cancelChan)
+	}
+
+	signer.Wg.Wait()
+
+	// Once the aux signer finishes handling the batch, the last job of the
+	// batch should have an empty response. Otherwise, the signer failed
+	// to skip that job after the cancel channel was closed.
+	select {
+	case <-jobs[numSigJobs-1].Resp:
+		t.Fatalf("Job cancellation failed")
+	default:
+	}
+}
+
+// TestAuxLeafSignerCancelAndQuit tests that the AuxLeafSigner will handle a
+// quit signal correctly, which involves ending sig job handling as soon as
+// possible. This test also sends a cancel signal before the quit signal, to
+// check that quits are handled correctly alongside other sent signals.
+func TestAuxLeafSignerCancelAndQuit(t *testing.T) {
+	cfg := &LeafSignerConfig{
+		ChainParams: testChainParams,
+		Signer:      &mockVirtualSigner{},
+	}
+
+	signer := NewAuxLeafSigner(cfg)
+	require.NoError(t, signer.Start())
+
+	defer func() {
+		require.NoError(t, signer.Stop())
+	}()
+
+	chanState := &channeldb.OpenChannel{
+		ChanType: channeldb.AnchorOutputsBit |
+			channeldb.ScidAliasChanBit | channeldb.SingleFunderBit |
+			channeldb.SimpleTaprootFeatureBit |
+			channeldb.TapscriptRootBit,
+		IsInitiator: true,
+	}
+	randInputProof := randProof(t)
+	commitTx := &randInputProof.AnchorTx
+	outgoingHtlcs := make(map[input.HtlcIndex][]*cmsg.AssetOutput)
+	outgoingHtlcs[0] = []*cmsg.AssetOutput{
+		cmsg.NewAssetOutput(
+			randInputProof.Asset.ID(), randInputProof.Asset.Amount,
+			randInputProof,
+		),
+	}
+
+	com := cmsg.NewCommitment(
+		nil, nil, outgoingHtlcs, nil, lnwallet.CommitAuxLeaves{},
+	)
+	cancelChan := make(chan struct{})
```

Comment:
Looks like we can consolidate the boiler plate above this into a shared func for both tests.